### ISSUE TITLE
Methods that should be protected in RateThread

### ISSUE THREAD

**Issue by drdanz** [robotology/yarp#1481]
The following methods in `yarp::os::RateThread` are supposed to be `protected`, but are `public`instead:

```c++
virtual bool threadInit();
virtual void threadRelease();
virtual void run() = 0;
virtual void beforeStart();
virtual void afterStart(bool success);
```

I'd like to make them `protected` but this is an API change and I don't know if this will break stuff around.
@traversaro @pattacini @lornat75 @barbalberto @vtikha are you aware of any module, device or other, actually *calling* these methods? (Just *overloading* them is ok, even though I would recommend to make them protected also in derived classes)

**Comment by pattacini** [comment]
@drdanz I don't believe there'll be any particular problem.
A quick test would be compiling `icub-main` with this change and see whether it'll be all fine.

**Comment by drdanz** [comment]
Cool, icub-main compiles, I'll open a PR...

**Comment by Nicogene** [comment]
Fixed by #1482
